extmod/modopenamp_rproc.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023-2024 Arduino SA
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * OpenAMP's remoteproc class.
+ */
+#if MICROPY_PY_OPENAMP_RPROC
```

Comment:
This should be `MICROPY_PY_OPENAMP_REMOTE_PROC` to match the class name `openamp.RemoteProc`.

Reply:
Should we just revert the class name to `RProc` ? Otherwise the files `modopenamp_rproc`, and `mprprocport` may need to be rename too. I will add the prefix `openamp_` to the type though.

Reply:
Everything is now renamed to `openamp_rproc_x`. Does the class name have to match ? I like `RemoteProc` better, but would still like to keep file, functions, macros names etc.. short (i.e `rproc`)

Reply:
> I like RemoteProc better

I agree. We should rename the files too. e.g. `openamp_remoteproc_x`.

Reply:
Okay, renamed everything including files @jimmo @dpgeorge